js/modules/k6/experimental/tracing/tracing.go

```diff
@@ -0,0 +1,278 @@
+		}
+	}
+
+	// Inject the updated HTTP module object in the runtime,
+	// overriding any previously imported one in the process.
+	err = t.vu.Runtime().Set("http", httpModuleObj)
+	if err != nil {
+		common.Throw(t.vu.Runtime(), err)
+	}
+}
+
+// configure configures the tracing module with the given options.
+func (t *Tracing) configure(opts instrumentationOptions) error {
+	if err := opts.validate(); err != nil {
+		return fmt.Errorf("invalid options: %w", err)
+	}
+
+	switch opts.Propagator {
+	case "w3c":
+		t.propagator = &W3CPropagator{}
+	case "b3":
+		t.propagator = &B3Propagator{}
+	case "jaeger":
+		t.propagator = &JaegerPropagator{}
+	default:
+		return fmt.Errorf("unknown propagator: %s", opts.Propagator)
+	}
+
+	return nil
+}
+
+// instrumentationOptions are the options that can be passed to the
+// tracing.instrument() method.
+type instrumentationOptions struct {
+	// Propagation is the propagation format to use for the tracer.
+	Propagator string `js:"propagator"`
+
+	// Sampling is the sampling rate to use for the tracer.
+	Sampling *float64 `js:"sampling"`
+
+	// Baggage is a map of baggage items to add to the tracer.
+	Baggage map[string]string `js:"baggage"`
+}
+
+func (i *instrumentationOptions) validate() error {
+	var (
+		isW3C    = i.Propagator == W3CPropagatorName
+		isB3     = i.Propagator == B3PropagatorName
+		isJaeger = i.Propagator == JaegerPropagatorName
+	)
+	if !isW3C && !isB3 && !isJaeger {
+		return fmt.Errorf("unknown propagator: %s", i.Propagator)
+	}
+
+	if i.Sampling != nil {
+		return fmt.Errorf("sampling is not yet supported")
+	}
+
+	if i.Baggage != nil {
+		return fmt.Errorf("baggage is not yet supported")
+	}
+
+	return nil
+}
+
+// instrumentHTTPMethod returns a new function that wraps the original http
+// method, adding tracing headers to the request.
+//
+// The function takes the HTTP method name as argument, as well as the exported
+// method itself, as a goja.Callable.
+//
+// The produced resulting function is ready to be injected back into the http
+// module, in place of the original method.
+func (t *Tracing) instrumentHTTPMethod(methodName k6HTTPMethodName, methodFn goja.Callable) goja.Callable {
+	return func(this goja.Value, args ...goja.Value) (goja.Value, error) {
+		rt := t.vu.Runtime()
+
+		// Ensure the arguments have a params object, in which
+		// we can add the tracing headers at a later point in time.
+		args, params, err := t.getOrCreateParams(methodName, args...)
+		if err != nil {
+			common.Throw(rt, fmt.Errorf("failed to normalize the params argument; reason: %w", err))
+		}
+
+		// Ensure that the params object contains a headers object.
+		// Create it if it doesn't.
+		headers, err := t.getOrCreateHeaders(params)
+		if err != nil {
+			common.Throw(rt, fmt.Errorf("failed to normalize params argument headers property; reason: %w", err))
+		}
+
+		traceID := NewTraceID(k6Prefix, k6CloudCode, uint64(time.Now().UnixNano())/uint64(time.Millisecond))
```

Comment:
You're right 🤔 

This effectively operates a conversion from nanoseconds to milliseconds. Was there an explicit reason for that @dgzlopes (this this statement is taken from xk6-distributed-tracing)? Maybe some precision requirements or considerations? Also, I've reused the `UnixTimestampNano` attribute definition, is there a reason for it to be a `uint64`, rather than a `int64`, like the result of `UnixNano()`?

Cheers 🙇🏻

Reply:
I'll be honest: I don't remember :smile: 

So yeah, probably no explicit reason.

Reply:
Hehe, alrighty, I'll just store nanoseconds, and as `int64` then 👍🏻

Reply:
@oleiade Please keep UnixTimestampNano as `uint64`. Otherwise, this will not work, as we will not be able to decode it on the Crocospans side.

Reply:
@Blinkuu, that's good to know, thanks for the info. I will keep it as a `uint64` then 👍🏻 🙇🏻